ThirdPartyAdapters/zucks/zucks/src/main/java/com/google/android/gms/ads/mediation/ZucksBannerAdapter.java

```
@@ -0,0 +1,342 @@
+        .requestBannerAd(
+            context,
+            mediationBannerListener,
+            serverParameters,
+            adSize,
+            mediationAdRequest,
+            mediationExtras);
+  }
+
+  @VisibleForTesting
+  @Nullable
+  AdError configureBannerAd(
+      Context context, AdSize adSize, Bundle serverParams, AdBannerListener listener) {
+    if (!isSizeSupported(adSize)) {
+      return ErrorMapper.createAdapterError(
+          ErrorMapper.ERROR_INVALID_REQUEST, "It is not a supported size. size=" + adSize);
+    }
+
+    String adFrameId = AdMobUtil.getFrameId(serverParams);
+
+    if (adFrameId == null) {
+      return ErrorMapper.createAdapterError(
+          ErrorMapper.ERROR_INVALID_REQUEST, "FrameID not contained in serverParameters.");
+    } else {
+      this.adSize = adSize;
+      zucksBanner = new AdBanner(context, adFrameId, listener);
+      AdMobUtil.configurePlatform(zucksBanner);
+      zucksBanner.load();
+      return null;
+    }
+  }
+
+  @Deprecated
+  @Override
+  public View getBannerView() {
+    return useBannerAdapter().getBannerView();
+  }
+
+  @NonNull
+  @Override
+  public View getView() {
+    return useBannerAd().getView();
+  }
+
+  @Deprecated
+  @Override
+  public void onDestroy() {
+    useBannerAdapter().onDestroy();
+  }
+
+  @Deprecated
+  @Override
+  public void onPause() {
+    useBannerAdapter().onPause();
+  }
+
+  @Deprecated
+  @Override
+  public void onResume() {
+    useBannerAdapter().onResume();
+  }
+
+  /** Validate passed size are supported in Zucks Ad Network SDK. */
+  @VisibleForTesting
+  static boolean isSizeSupported(AdSize adSize) {
+    return adSize.equals(AdSize.BANNER)
+        || adSize.equals(new AdSize(320, 50))
+        || adSize.equals(AdSize.LARGE_BANNER)
+        || adSize.equals(AdSize.MEDIUM_RECTANGLE);
+  }
+
+  @VisibleForTesting
+  @Nullable
+  AdError isValidAdSize(@NonNull AdBanner banner) {
+    return isValidAdSize(adSize, banner);
+  }
```

Can this be removed?

---

Hmm, Why did think so? I may not be aware of the ideal code.

`ZucksBannerAdapter#isValidAdSize` is internal (non-static) method for asserts  AdMob's banner (adapter) size and Zucks's pre-defined ad unit size are equals. It's fired as an instance's self state validation in the adapter lifecycle.

Implement this method, require knowing the internal property of ZucksBannerAdapter. and need to return an error if size is invalid. therefore method creation is suitable I think.

`ZucksBannerAdapter::isValidAdSize` is it's actual implementation. This is split for unit testing. Considering how its used, the former (non-static) method is suitable. But considering how to testing logics, this  ​(static) method is more suitable